Geometry/HGCalCommonData/plugins/DDHGCalEEAlgo.cc

```diff
@@ -204,12 +206,26 @@ void DDHGCalEEAlgo::constructLayers(const DDLogicalPart& module,
       DDLogicalPart glog;
       if (layerSense_[ly] < 1) {
 	double alpha = CLHEP::pi/sectors_;
-	int    nsec  = (layerSense_[ly] == 0 || absorbMode_ == 0) ? 2 : 2;
+	int    nsec  = (layerSense_[ly] == 0 || absorbMode_ == 0 ||
+			zr.first < 0) ? 2 : 3;
 	std::vector<double> pgonZ(nsec), pgonRin(nsec), pgonRout(nsec);
-	double rmax  = routF*cos(alpha) - tol;
-	pgonZ[0]    =-hthick;  pgonZ[1]    = hthick;
-	pgonRin[0]  = rinB;    pgonRin[1]  = rinB;   
-	pgonRout[0] = rmax;    pgonRout[1] = rmax;   
+	if (layerSense_[ly] == 0 || absorbMode_ == 0) {
+	  double rmax  = routF*cos(alpha) - tol;
```

Comment:
since `alpha` is effectively a constant (`sectors_` is an input parameter that doesn't change), this trig computation `cos(alpha)` can be defined once, outside of the loops in this function, to avoid repeating the computation (also in the other algorithms below)

Reply:
Done

Reply:
Taken care of